test/generators/shop_model_generator_test.rb

```diff
@@ -29,6 +33,13 @@ class ShopModelGeneratorTest < Rails::Generators::TestCase
     end
   end
 
+  test "injects into application controller" do
+    run_generator
+    assert_file "app/controllers/application_controller.rb" do |controller|
+      assert_match "def shop", controller
+    end
```

Comment:
oh shoot - this isn't merged. You'll need to add the shop method. Sorry for the confusion. The ID is stored in the session so you can write a method to fetch the model using the ID.

We should probably merge this but we need to figure out if we are making the Shop model part of this Engine as non optional or not.